rstudio/rsconnect-python
# implement publishing static outputs for cloud

## Intent





Adds support for `deploy html` and `deploy manifest` to deploy to Posit Cloud.

Resolves https://github.com/rstudio/rsconnect-python/issues/420.

This includes the changes from https://github.com/mslynch/rsconnect-python/pull/1, plus versioned app store files and more tests.

## Type of Change




## Approach



Allows for static content to be deployed to Posit Cloud.

This PR also introduces file versioning to the app store files with an initial version of 1. For cloud deployments, `app_id` will now store the content id (as in `https://posit.cloud/content/{content_id}` (as opposed to the previous approach of storing the application id).

Also contains a refactor to Cloud's HTTP client to handle errors within the client and return the parsed response body from its calls.

## Automated Tests



Added a test with `rsconnect deploy manifest` using a manifest for a `static`-appmode application. Also added several unit tests for `CloudService.prepare_deploy` to test deploy/redeploy/pre-versioned app store files.

## Directions for Reviewers


To test this manually, you can use:
```shell
rsconnect deploy html \
          --account $YOUR_ACCOUNT \
          --token $YOUR_TOKEN \
          --secret $YOUR_SECRET \
          --server 'https://api.staging.posit.cloud' \
          'some_file.html'
```

To test compatibility with pre-versioned app store files,
- do a deploy on the current version:
  ```shell
  rsconnect deploy shiny \
            --account $YOUR_ACCOUNT \
            --token $YOUR_TOKEN \
            --secret $YOUR_SECRET \
            --server 'https://api.staging.posit.cloud' \
            'my-shiny-app'
  ```
- then check out main, reinstall, and run the command again.

## Checklist


- [ ] I have updated all related GitHub issues to reflect their current state.
